builtin/logical/database/path_rotate_credentials.go

```diff
@@ -144,7 +149,19 @@ func (b *databaseBackend) pathRotateRoleCredentialsUpdate() framework.OperationF
 			return nil, err
 		}
 
-		return nil, nil
+		// Send a canary to force a guard check which prevents the client
```

Comment:
I'm not exactly following what the issue is here, if there is no write then why would the secondary be waiting for a WAL value and timing out?  Also doesn't the above call to `setStaticAccount` write to storage, making this block unnecessary?

Reply:
Thanks, @briankassouf. I did some more investigation and my initial assumption was wrong. The real issue exists in the forwarding mechanism from the replication system. I removed that part.